client.go

```diff
@@ -163,12 +173,17 @@ type EncryptResult struct {
 
 // Encrypt encrypts the plaintext using the specified algorithm.
 func (client *Client) Encrypt(ctx context.Context, algorithm EncryptAlgorithm, plaintext []byte, options *EncryptOptions) (EncryptResult, error) {
-	client.init(ctx)
+	var err error
+	ctx, span := client.startSpan(ctx, "Encrypt")
+	defer func() { span.End(err) }()
+
+	client.cache(ctx)
 
 	var encrypter alg.Encrypter
 	if alg.As(client.localClient, &encrypter) {
 		result, err := encrypter.Encrypt(algorithm, plaintext)
```

Comment:
`err` is being shadowed here so `span.End()` isn't going to report it (I believe there are other instances of this).

Reply:
Not if it was already defined: https://go.dev/play/p/A9Kw41FwWXX

I believe this only works because it was not created in a block expression. I remember coming across this in a doc long ago, but having trouble finding it. Still, if you think it's better to declare `result` explicit then use `=` to assign both, I could switch to that.

Reply:
Try [this](https://go.dev/play/p/N8MwUs0sWEX).  It fails to compile because `err` is declared but not used.

Reply:
I see. So scope matters. I've usually only used that in the same scope. This isn't well-documented, it seems. 😔